drivers/syslog/syslog_channel.c

```diff
@@ -259,13 +259,26 @@ int syslog_channel(FAR struct syslog_channel_s *channel)
   int i;
 #endif
 
+#if defined(CONFIG_SYSLOG_COLOR_OUTPUT)
+  char * col_rst = "\e[0m";
+#endif
+
   DEBUGASSERT(channel != NULL);
 
   if (channel != NULL)
     {
       DEBUGASSERT(channel->sc_ops->sc_putc != NULL &&
                   channel->sc_ops->sc_force != NULL);
 
+#if defined(CONFIG_SYSLOG_COLOR_OUTPUT)
+      /* Reset the terminal style before using it. */
```

Comment:
what about the default channel?

Reply:
Indeed, I didn't think about this.

This is not very straight-forward, since the default channel is statically initialized.  
How do you propose to handle this?

Reply:
Can we move https://github.com/apache/incubator-nuttx/blob/master/drivers/syslog/vsyslog.c#L218-L222 to the beginning?

Reply:
This is one possible solution, but it will always leave the terminal configured on the last used color, instead of returning it to normal like it does now.

Is this desirable?

I think this will cause unwanted side-effects.  
Examples:
* Using the same serial port both for the logger and for NSH.
* When restarting the system, a bootloader may need to use the serial output before NuttX starts (and resets the terminal).

Another solution is to add the color sequence both at the start and the end of the print, but it seems a bit excessive.

Reply:
The BL might not use color output at all. I mean that we can't handle all the cases. I'm not sure even that Linux and other Unix systems does. For example recently I've been using Windows + cmake in terminal and when cmake compilation failed the terminal in Windows was left at the color of cmake stage that failed. I do not see any was how we can handle all the corner cases.

Reply:
I'm not sure if we need this change, as seems that only the first message will be printed in a wrong color or if no syslog messages (very unlikely) are in place then the terminal will remain in wrong color.